server/neptune/workflows/internal/deploy/revision/queue/deployer.go

```diff
@@ -86,12 +91,48 @@ func (p *Deployer) Deploy(ctx workflow.Context, requestedDeployment terraformWor
 		logger.Error(ctx, "unable to persist deployment, proceeding with in-memory value", key.ErrKey, persistErr)
 	}
 
+	// rebase open PRs
+	// worker uses the returned error types to perform some follow up tasks, so instead of propagating the rebase error back, we configure
+	// infinite retries and log the error to ensure we rebase open PRs before deploying another change in this root
+	// ScheduleToCloseTimeout is configured to 8 hours to ensure we're not blocking deploys for this root if the issue persists
+	if rebaseErr := p.rebaseOpenPRsForRoot(ctx, requestedDeployment.Repo); rebaseErr != nil {
+		scope.Counter("rebase_open_prs_err").Inc(1)
+		logger.Error(ctx, "unable to rebase open PRs", key.ErrKey, rebaseErr)
+	}
+
 	// Count this as deployment as latest if it's not a PlanRejectionError which means it is a TerraformClientError
 	// We do this as a safety measure to avoid deploying out of order revision after a failed deploy since it could still
 	// mutate the state file
 	return latestDeployment, err
 }
 
+func (p *Deployer) rebaseOpenPRsForRoot(ctx workflow.Context, repo github.Repo) error {
+	// skip rebasing open PRs if the execution is still on the old workflow version
+	version := workflow.GetVersion(ctx, version.RebaseOpenPRs, workflow.DefaultVersion, 1)
+	if version == workflow.DefaultVersion {
+		return nil
+	}
+
+	// configure infinite retries and ScheduleToCloseTimeout to 8 hours to allow for the GH API Ratelimit to revive if we hit the ratelimit since it resets every hour
```

Comment:
I don't really understand what this means, why do we need a custom scheduletoclosetimeout? This means that we will block deploys if this fails and i don't think we should be doing that at all.

If anything i think all this logic can be executed as async activities and shouldn't block at all.

Reply:
Agreed! What do you think of starting a new temporal goroutine to do the rebase flow so that deploys are not blocked while rebase is in progress?